### PR TITLE
Material design

### PR DESCRIPTION
**Pull request by BRHoward**
- Changed styling to material design
- Different conversations are displayed through tabs
- Guests can now set a custom avatar on sign up

### CODE REVIEW

**Comment by BRHoward** [comment]
@jebbinSCL @stevenhillcox @sievins

**Comment by jebbinSCL** [comment]
Your public/index.html is looking quite big. Can it be broken down to sub templates?

**Comment by jebbinSCL** [comment]
The main issue is the size of your index.html. If this will cause merge issues will a pull request you have waiting, you can do it afterwards, otherwise, you need to break it down. Other than that things are fine.

**Comment by BRHoward** [comment]
@jebbinSCL

File: public/js/ChatController.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-    /* global _, angular */
+    /* global _, angular, console */
```
Comment (stevenhillcox-sl) [inline public/js/ChatController.js]: You shouldn't be using console in your actual code. I assume this is left over from debugging.

File: public/js/ChatController.js [hunk 2]
```diff
@@ -146,6 +156,35 @@
...
+                return "Loading...";
+            }
+            var otherUsersIds = convo.userids.filter(function (id) {
+                return $scope.currentUserData.id !== id;
+            });
+            if (otherUsersIds.length === 0) {
+                return "Just you";
+            }
+            var outputString = "";
+            outputString += getUserFromId(otherUsersIds[0]).name;
```
Comment (stevenhillcox) [inline public/js/ChatController.js]: Consider using a filter here.

File: public/js/ChatController.js [hunk 3]
```diff
@@ -146,6 +156,35 @@
...
+            if (otherUsersIds.length > 1) {
+                outputString += " and ";
+                outputString += otherUsersIds.length - 1;
+                outputString += " other";
+            }
+            if (otherUsersIds.length > 2) {
```
Comment (stevenhillcox-sl) [inline public/js/ChatController.js]: Consider using a filter here

File: server/server.js [hunk 4]
```diff
@@ -66,12 +66,16 @@ module.exports = function (port, db, githubAuthoriser) {
 
     app.post("/api/newGuest", function (req, res) {
         var guestID = uuid.v4();
-        var guestName = req.body.name;
+        var avatarUrl = req.body.avatarUrl;
+        //if the user didn't give an avatar url then give a default one
+        if (avatarUrl === "") {
```
Comment (stevenhillcox-sl) [inline server/server.js]: Consider using !avatarUrl to cover undefined and null cases.

**Comment by BRHoward** [comment]
@jebbinSCL  @stevenhillcox

File: public/js/ChatController.js [hunk 5]
```diff
@@ -65,15 +67,17 @@
...
-            RequestService.postGuestLogin($scope.nameInputBox)
-                .then(function (response) {
-                    $scope.loadUserInfo();
-                    $scope.errorText = "";
-                }, function (response) {
-                    $scope.errorText =
-                        "Failed to login as guest : " + response.status + " - " + response.statusText;
-                });
+        function guestLogin(name, avatarURL) {
+            if (name !== "") {
```
Comment (jebbinSCL) [inline public/js/ChatController.js]: Same issue that Steven pointed out. Thinking about using !name or another alternative.

File: public/js/ChatController.js [hunk 6]
```diff
@@ -146,6 +156,20 @@
                 });
         }
 
+        function getConversationLabel(conversationid) {
+            //generates the string to be shown on the conversation tabs
+            //either shows the conversation name or gives an indication of
+            //how many people are in the conversation
+            var convo = getConversationFromId(conversationid);
+            if (convo.name !== "") {
```
Comment (jebbinSCL) [inline public/js/ChatController.js]: Same as above.

**Comment by jebbinSCL** [comment]
:ship:

**Comment by stevenhillcox-sl** [comment]
Looks fine to me

**Pull request merged**